alpaca_trade_api/rest.py

```diff
@@ -612,29 +612,38 @@ def _data_get(self,
                   symbol_or_symbols: Union[str, List[str]],
                   api_version: str = 'v2',
                   endpoint_base: str = 'stocks',
+                  resp_grouped_by_symbol: bool = None,
                   **kwargs):
         page_token = None
         total_items = 0
         limit = kwargs.get('limit')
+        page_limit = DATA_V2_MAX_LIMIT
+        if endpoint_base == "news":
+            page_limit = NEWS_MAX_LIMIT
         while True:
             actual_limit = None
             if limit:
-                actual_limit = min(int(limit) - total_items, DATA_V2_MAX_LIMIT)
+                actual_limit = min(int(limit) - total_items, page_limit)
                 if actual_limit < 1:
                     break
             data = kwargs
             data['limit'] = actual_limit
             data['page_token'] = page_token
-            if isinstance(symbol_or_symbols, str):
-                path = f'/{endpoint_base}/{symbol_or_symbols}/{endpoint}'
-            else:
-                path = f'/{endpoint_base}/{endpoint}'
+            path = f'/{endpoint_base}'
+            if isinstance(symbol_or_symbols, str) and symbol_or_symbols:
+                path += f'/{symbol_or_symbols}'
+            if endpoint:
+                path += f'/{endpoint}'
+            if isinstance(symbol_or_symbols, list) and symbol_or_symbols:
```

Comment:
You're right. But I think this okay because the type hint is `symbol_or_symbols: Union[str, List[str]]`. So it's not breaking any promises. I'm leaving it if it's okay with you too

Reply:
Well, type hints are just hints in python. Technically, this is a breaking change.

Reply:
Changed to previous condition